README.md

```diff
@@ -154,14 +154,18 @@ PLUGINS_CONFIG = {
     "nautobot_secrets_providers": {
         "hashicorp_vault": {
             "url": "http://localhost:8200",
+            "auth_method": "token",
             "token": os.getenv("NAUTOBOT_HASHICORP_VAULT_TOKEN"),
         }
     },
 }
 ```
 
-- `url` - The URL to the HashiCorp Vault instance (e.g. `http://localhost:8200`)
-- `token` - The token for authenticating the client with the HashiCorp Vault instance. As with other sensitive service credentials, we recommend that you provide the token value as an environment variable and retrieve it with `{"token": os.getenv("NAUTOBOT_HASHICORP_VAULT_TOKEN")}` rather than hard-coding it in your `nautobot_config.py`.
+- `url` - (required) The URL to the HashiCorp Vault instance (e.g. `http://localhost:8200`).
+- `auth_method` - (optional / defaults to "token") The method used to authenticate against the HashiCorp Vault instance.
+- `token` - (optional) Required when `"auth_method": "token"` or auth_method is not supplied. The token for authenticating the client with the HashiCorp Vault instance. As with other sensitive service credentials, we recommend that you provide the token value as an environment variable and retrieve it with `{"token": os.getenv("NAUTOBOT_HASHICORP_VAULT_TOKEN")}` rather than hard-coding it in your `nautobot_config.py`.
+- `role_id` - (optional) Required when `"auth_method": "approle"`.
+- `secret_id` - (optional) Required when `"auth_method": "approle"`.
```

Comment:
If either or both of these are sensitive values, should we have the same disclaimer about using environment variables vs hard-coding?

Also, not necessarily in scope for this PR, but I know it'll get asked - could we have an alternative configuration method where `token`, `role_id`, and/or `secret_id` could be specified as slugs of Nautobot secrets to use to look up the actual values? I.e. Nautobot could then pull the `token`, `role_id`, or `secret_id` from any (non-Vault) backend that it supports, whether that be an environment variable, text file, Thycotic, etc...